TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ftc16072/OpModes/TestWiring.java

```diff
@@ -0,0 +1,79 @@
+package org.firstinspires.ftc.teamcode.ftc16072.OpModes;
+
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.teamcode.ftc16072.Mechanisms.Mechanism;
+import org.firstinspires.ftc.teamcode.ftc16072.QQTest.QQtest;
+import org.firstinspires.ftc.teamcode.ftc16072.Robot;
+
+import java.util.List;
+
+@TeleOp
+public class TestWiring extends OpMode {
+    Robot robot = new Robot();
+    List<Mechanism> mechanismList;
+    int currentTest;
+    int currentMechanism;
+    List<QQtest> testList;
+
+
+    public void init(){
+        mechanismList = robot.getMechanismList();
+
+    }
+    public void loop(){
+        if (gamepad1.dpad_up){ //swap mechanisms
+            // alternate 1 mechanism up unless you are already at the last one in which you move to the first one
+            if (currentMechanism >= mechanismList.size()){
+                currentMechanism = 0;
+            }
+            else {
+                currentMechanism = currentMechanism + 1;
+
+            }
+            currentTest = 0;
+
+            // update mechanism
+
+        }
+        else if (gamepad1.dpad_down){ // possible failure point
+            if (currentMechanism <= 0){
+                currentMechanism = mechanismList.size();
```

Comment:
Should be:
`currentMechanism = mechanismList.size() - 1;`

Reply:
Setting to mechanismList.size() will cause it to go past the end of the array because arrays are indexed starting with zero.